angular-register.ts

```diff
@@ -0,0 +1,30 @@
+declare interface NoopDecorator {
+  <T>(type: T): T;
```

Comment:
Why do we need this prototype?

Reply:
This looks like a class decorator, but `@Override` is not meant to be applied on a class.

Reply:
The idea here was to provide proper VS Code popups.

Example with interface:
![Screen Shot 2020-01-16 at 19 01 40](https://user-images.githubusercontent.com/8206108/72575664-c7ff4f00-3892-11ea-9fe1-ce541a8967bd.png)

Example with type:
![Screen Shot 2020-01-16 at 19 01 58](https://user-images.githubusercontent.com/8206108/72575678-d0f02080-3892-11ea-82bf-f99c242a80c8.png)

Leaving off the ` <T>(type: T): T;` results in TSLint converting this back to a type and resulting in imho a less meaningful tool tip.
![Screen Shot 2020-01-16 at 19 04 13](https://user-images.githubusercontent.com/8206108/72575751-03018280-3893-11ea-8fe6-793fc819b4de.png)

If you want to use a `//tslint-disable` on this line that would be good too.

Reply:
Removed generic